extensions/wikia/PageHeader/PageTitle.class.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace PageHeader;
+
+class PageTitle {
+
+    /* @var string */
+    private $title;
+
+    /* @var \WikiaGlobalRegistry */
+    private $wg;
+
+    /* @var int */
+    private $namespace;
+
+    /**
+     * @param \WikiaApp $app
+     */
+    public function __construct($app) {
+        $this->wg = $app->wg;
+
+        $this->title = $app->getSkinTemplateObj()->data['title'];
+        $this->namespace = $this->wg->Title->getNamespace();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)